src/main/java/com/example/hop_oasis/service/data/CiderImageServiceImpl.java

```diff
@@ -22,44 +24,40 @@
 @RequiredArgsConstructor
 public class CiderImageServiceImpl implements CiderImageService {
     private final CiderImageRepository ciderImageRepository;
-    private final CiderImageMapper ciderImageMapper;
-    private final ImageCompressor imageCompressor;
     private final CiderRepository ciderRepository;
-
+    private final S3Service s3Service;
+    private final CiderInfoMapper ciderInfoMapper;
+    private final ExtractImageName extractImageName;
     @Override
-    public CiderImageDto getCiderImageByName(String name) {
-        Optional<CiderImage> imageOp = ciderImageRepository.findFirstByName(name);
-        if (imageOp.isEmpty()) {
-            throw new ResourceNotFoundException(RESOURCE_NOT_FOUND, name);
-        }
-        CiderImage image = imageOp.get();
-        image.setImage(imageCompressor.decompressImage(image.getImage(), name));
-        return ciderImageMapper.toDto(image);
+    public ImageUrlDto getCiderImageByName(String name) {
+     return new ImageUrlDto(s3Service.getFileUrl(name).toString());
     }
     @Override
-    public CiderImage addCiderImageToCider(Long ciderId, MultipartFile file) {
-        byte[] image;
+    public CiderInfoDto addCiderImageToCider(Long ciderId, MultipartFile file) {
         try {
-            image = imageCompressor.compressImage(file.getBytes());
+            String fileName ="ciders/" + file.getOriginalFilename();
+            s3Service.uploadFile(fileName, file);
         } catch (IOException e) {
             throw new ResourceNotFoundException(RESOURCE_NOT_FOUND, "");
         }
         CiderImage image1 = CiderImage.builder()
-                .image(image)
-                .name(file.getOriginalFilename())
+                .name(s3Service.getFileUrl(file.getOriginalFilename()).toString())
                 .build();
         Cider cider = ciderRepository.findById(ciderId).orElseThrow(() ->
                 new ResourceNotFoundException(RESOURCE_NOT_FOUND, ciderId));
 
         image1.setCider(cider);
-       return ciderImageRepository.save(image1);
+        ciderImageRepository.save(image1);
+       return ciderInfoMapper.toDto(ciderRepository.findById(ciderId).orElseThrow(() ->
```

Comment:
the very thing you should do is to check if a cider exists. Imagine you did all this stuff, uploaded image to s3 and then figured that image with this id doesn't exist, and throw an Exception.
All validation must be performed at the beginning of the method.